### PR TITLE
Add flags about whether new rooms are encrypted by default.

### LINKED ISSUE #8311: [20200921] Expose forced-on E2EE through a /versions flag or something
Capabilities are probably wrong for this, and we don't yet have an info API. For speed, I'd recommend a `/versions` flag of `io.element.e2ee_forced` for clients to check.

### PR DESCRIPTION
**Pull request by clokep**
Fixes #8311

CC @turt2live

### CODE REVIEW

File: synapse/rest/client/versions.py [hunk 1]
```diff
@@ -31,6 +32,18 @@ def __init__(self, hs):
...
 
+        # Calculate these once since they shouldn't change after start-up.
+        self.e2ee_forced_public = (
+            RoomCreationPreset.PUBLIC_CHAT
+            in self.config.encryption_enabled_by_default_for_room_presets
+        )
+        self.e2ee_forced_private = (
+            RoomCreationPreset.PRIVATE_CHAT
+            in self.config.encryption_enabled_by_default_for_room_presets
+            or RoomCreationPreset.TRUSTED_PRIVATE_CHAT
```
Comment (clokep) [inline synapse/rest/client/versions.py:43]: I wondered if this should be `and` instead of `or`...

Comment (turt2live) [reply synapse/rest/client/versions.py:43]: I think neither? https://github.com/matrix-org/synapse/blob/837293c314b47e988fe9532115476a6536cd6406/docs/sample_config.yaml#L2176-L2191 has options of `all`, `invite`, or `off` (default)

Comment (clokep) [reply synapse/rest/client/versions.py:43]: In the case `invite`, the config ends up being `["private_chat", "trusted_private_chat"]`.

All is the two above + `"public_chat"`.

(Off it ends up being empty: `[]`.)

Comment (turt2live) [reply synapse/rest/client/versions.py:43]: exciting. Probably `and` in that case, for added safety

Comment (clokep) [reply synapse/rest/client/versions.py:43]: Or we can add a third item as `trusted_private`?

Comment (turt2live) [reply synapse/rest/client/versions.py:43]: sure, whatever is easiest :D

Comment (clokep) [reply synapse/rest/client/versions.py]: I separated it out, this should let the client tweak behavior without the server needing changes.

**Comment by turt2live** [comment]
This works great btw :D

**Review by erikjohnston — APPROVED** [review]

**Comment by clokep** [comment]
> This works great btw :D

Yay! We're going to put this into v1.20.0rc5.

**Pull request merged**